### ISSUE TITLE
Autocomplete nickname when colon is attached to prefix

### ISSUE THREAD

**Issue by jeffreyroberts** [weechat/weechat#1210]
## Bug, feature, or question?

feature

## Current behavior

nic:

results in

nic:

## Desired behavior

would be dope if it did autocomplete like BitchX

nickname:

**Comment by weechatter** [comment]
why not pressing [TAB] instead of colon?
nic[tab] should autocomplete to nickname:

**Comment by jeffreyroberts** [comment]
BitchX does both, Tab for autocomplete as well as : 

I personally just made the change from BitchX to weechat today, im sure others will appreciate this feature as well =]

**Comment by weechatter** [comment]
What does BitchX do when there are multiple nicks?
eg: nic1 , nic2 , nic3
and you do nic: 

complete with last spoken?

IMO this could be also done with a script as well

**Comment by jeffreyroberts** [comment]
[11:17AM] <jlroberts> remo, efu:

It appears BitchX doesn't do anything when there are multiple nicknames behind the colon

**Comment by judwhite** [comment]
@jeffreyroberts I think @weechatter means what if you do "nic:" and you have "niceguy", "nicotine", "nicknacks" and "nickelback" in the channel. I don't know about BitchX, but X-Chat when using tab completes the first name alphabetically, and hitting tab again will cycle through the list of nicks with the prefix you had entered.

**Comment by jjdredd** [comment]
I also think autocompletion for nicknames is a very useful feature.
I might look into it myself

Should autocompletion work everywhere in the input message, or just when you are starting to type the message, and not
somewhere in the middle?

**Comment by weechatter** [comment]
there are a couple of options for completion, like a "inline" completion, see /fset weechat.completion

**Comment by jjdredd** [comment]
I have found several var tha control nick completion.

```
weechat.completion.nick_completer
weechat.completion.nick_first_only
weechat.completion.nick_ignore_chars
```
Does this mean that nick completion feature is already implemented?

**Comment by trygveaa** [comment]
@jjdredd: Yes, WeeChat has nick completion. What this issue is asking for is being able to complete a nick after typing a partial nick and `:`. If you don't type the `:` it works fine.

**Comment by jjdredd** [comment]
Oh, my bad.

**Comment by weechatter** [comment]
@jjdredd : nick completion is part of core and implemented since begin of time :-D

jeffreyroberts asked for a function when you "nic:" (double point instead of TAB) for autocomplete. For me, a bit pointless, because you have TAB and it will autocomplete nick and add a char behind the nick (weechat.completion.nick_completer). 

Therefore i asked what happen when you "nic:" and the nicklist has different matches like "niclas", "nicolaus", nice". in weechat you can cycle through the nicks.
